File: app/src/test/java/fr/nspu/dev/recordvoicelearning/utils/QuestionToAnswerEnumTest.java
package fr.nspu.dev.recordvoicelearning.utils;

import org.junit.Test;

import static org.junit.Assert.*;

/**
 * Created by nspu on 24/02/18.
 */
public class QuestionToAnswerEnumTest {
    @Test
    public void toBoolean() throws Exception {
        boolean bTrue = QuestionToAnswerEnum.QUESTION_TO_ANSWER.toBoolean();
        boolean bFalse = QuestionToAnswerEnum.ANSWER_TO_QUESTION.toBoolean();

        assertEquals(bTrue, true);
        assertEquals(bFalse, false);
    }

    @Test
    public void toQuestionToAnswerEnum() throws Exception {
        QuestionToAnswerEnum questionToAnswerEnumTrue = QuestionToAnswerEnum.Companion.toQuestionToAnswerEnum(true);
        QuestionToAnswerEnum questionToAnswerEnumFalse = QuestionToAnswerEnum.Companion.toQuestionToAnswerEnum(false);
        
        assertEquals(questionToAnswerEnumTrue, QuestionToAnswerEnum.QUESTION_TO_ANSWER);
        assertEquals(questionToAnswerEnumFalse, QuestionToAnswerEnum.ANSWER_TO_QUESTION);
    }

}

File: app/src/test/java/fr/nspu/dev/recordvoicelearning/utils/QuestionToAnswerEnumTest.kt
package fr.nspu.dev.recordvoicelearning.utils

import org.junit.Assert.assertEquals
import org.junit.Test

/**
 * Created by nspu on 24/02/18.
 */
class QuestionToAnswerEnumTest {
    @Test
    @Throws(Exception::class)
    fun toBoolean() {
        val bTrue = QuestionToAnswerEnum.QUESTION_TO_ANSWER.toBoolean()
        val bFalse = QuestionToAnswerEnum.ANSWER_TO_QUESTION.toBoolean()

        assertEquals(bTrue, true)
        assertEquals(bFalse, false)
    }

    @Test
    @Throws(Exception::class)
    fun toQuestionToAnswerEnum() {
        val questionToAnswerEnumTrue = QuestionToAnswerEnum.toQuestionToAnswerEnum(true)
        val questionToAnswerEnumFalse = QuestionToAnswerEnum.toQuestionToAnswerEnum(false)

        assertEquals(questionToAnswerEnumTrue, QuestionToAnswerEnum.QUESTION_TO_ANSWER)
        assertEquals(questionToAnswerEnumFalse, QuestionToAnswerEnum.ANSWER_TO_QUESTION)
    }

}